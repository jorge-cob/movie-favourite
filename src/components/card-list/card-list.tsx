import React from 'react';

import { Card } from '../card/card';

interface Movie {
  title: string;
  releaseDate: string;
  overview: string;
  posterPath: string;
  id: number;
}

interface MovieListProps {
  movies: Movie[];
}

export const CardList = ({movies}: MovieListProps) => (
  <div className='card-list'>
    {movies.map(movie => (
      <Card key={movie.id} movieDetails={movie} />
    ))}
  </div>
);
