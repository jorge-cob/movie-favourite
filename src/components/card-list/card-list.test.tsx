import React from 'react';
import { shallow } from 'enzyme';
import { CardList } from './card-list';

const mockMovies = [
  {
    id: '522',
    title: 'Titanic',
    release_date: '1991-04-1',
    vote_average: '7.8',
    poster_path: 'http://fakeroute.com/titanic.jpg',
    overview: 'Wow, Jack, dont sink!',
  },
  {
    id: '362',
    title: 'Spider Man',
    release_date: '1976-04-12',
    vote_average: '7.8',
    poster_path: 'http://fakeroute.com/spide.jpg',
    overview: 'Wow, Jack, dont sink!',
  },
];

describe('CardList', () => {

  it('should render CardList component', () => {
    const component = shallow(<CardList movies={mockMovies} />)
    expect(component).toMatchSnapshot();
  });
  it('should render CardList component with no movies', () => {
    const component = shallow(<CardList movies={[]} />)
    expect(component).toMatchSnapshot();
  });
});