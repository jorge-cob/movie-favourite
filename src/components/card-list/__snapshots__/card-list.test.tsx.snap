// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should render CardList component 1`] = `ShallowWrapper {}`;

exports[`CardList should render CardList component 1`] = `ShallowWrapper {}`;

exports[`CardList should render CardList component with no movies 1`] = `ShallowWrapper {}`;

exports[`should render CardList component 1`] = `ShallowWrapper {}`;

exports[`should render CardList component with no movies 1`] = `ShallowWrapper {}`;
